2025-06-20 20:45:20.051 | INFO     | app:lifespan:86 - Initializing debug interface database...
2025-06-20 20:45:20.064 | INFO     | database:init_db:72 - Database initialized successfully
2025-06-20 20:45:20.065 | INFO     | app:lifespan:88 - Debug interface ready!
2025-06-20 20:45:43.090 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-20 20:45:43.091 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-20 20:45:43.357 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-20 20:45:43.366 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-20 20:45:49.041 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-20 20:45:49.043 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-20 20:45:49.072 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-20 20:45:49.083 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-21 13:52:10.884 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-21 13:52:10.916 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-21 13:52:11.085 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-21 13:52:11.145 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-21 14:02:51.646 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-21 14:02:51.648 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-21 14:02:51.696 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-21 14:02:51.706 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-21 14:02:57.190 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-21 14:02:57.191 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-21 14:02:57.220 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-21 14:02:57.230 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-21 14:13:41.525 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-21 14:13:41.535 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-21 14:13:41.575 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-21 14:13:41.589 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-21 14:13:46.987 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-21 14:13:46.988 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-21 14:13:47.006 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-21 14:13:47.014 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-21 14:20:49.214 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-21 14:20:49.221 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-21 14:20:49.266 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-21 14:20:49.293 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-21 14:20:50.466 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-21 14:20:50.469 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-21 14:20:50.491 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-21 14:20:50.500 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-21 14:20:54.537 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-21 14:20:54.538 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-21 14:20:54.560 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-21 14:20:54.569 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-21 14:20:55.933 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-21 14:20:55.934 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-21 14:20:55.952 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-21 14:20:55.961 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-21 14:21:00.301 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-21 14:21:00.301 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-21 14:21:00.326 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-21 14:21:00.334 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-22 02:08:25.467 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-22 02:08:25.515 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-22 02:08:25.893 | INFO     | debug_engine:debug_walk_section:607 - Debug engine response | completed=False question={'questionId': 'Q_AD_First_Name'} nextSection=None debug_steps=2
2025-06-22 02:08:25.961 | INFO     | app:debug_execute_flow:295 - Debug execution completed
2025-06-22 02:11:09.603 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-22 02:11:09.607 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-22 02:11:09.655 | ERROR    | app:debug_execute_flow:324 - Debug execution failed
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 120, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 352
               │     └ 3
               └ <function _main at 0x0000028C947D6480>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 133, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 352
           │    └ <function BaseProcess._bootstrap at 0x0000028C9427FC40>
           └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000028C9427F1A0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x0000028C9477EC10>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    └ <function subprocess_started at 0x0000028C96917BA0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x0000028C96A7C810>>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
           │       │   │    └ <function Server.serve at 0x0000028C96916C00>
           │       │   └ <uvicorn.server.Server object at 0x0000028C96A7C810>
           │       └ <function run at 0x0000028C9614A340>
           └ <module 'asyncio' from 'C:\\Program Files\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x0000028C96A808C0>
           │      └ <function Runner.run at 0x0000028C96264680>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\serve...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000028C96262200>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 637, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x0000028C96262160>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 604, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x0000028C96263F60>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000028C9612E520>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Program Files\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBE...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBE...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBE...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
          └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBC3C40>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
          │    │               │      │        │                     └ Headers({'host': 'localhost:8005', 'connection': 'keep-alive', 'content-length': '170', 'sec-ch-ua-platform': '"Windows"', 'u...
          │    │               │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBC3C40>
          │    │               │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │               └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function CORSMiddleware.simple_response at 0x0000028C98B604A0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
          │    │   │      │        └ functools.partial(<bound method CORSMiddleware.send of <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050...
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBC3920>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBC3920>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x0000028C941943D0>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBC3920>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │     └ <function Route.handle at 0x0000028C98AC84A0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBC3920>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function request_response.<locals>.app at 0x0000028CBEB76FC0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x0000028CBEC0ED90>
                     └ <function get_request_handler.<locals>.app at 0x0000028CBEB76F20>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x0000028C98A92520>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='...
                 │         └ <function debug_execute_flow at 0x0000028CBEB76160>
                 └ <fastapi.dependencies.models.Dependant object at 0x0000028CBEBA9D50>

> File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\app.py", line 279, in debug_execute_flow
    response, debug_info = debug_walk_section(request.sectionId, ctx_dict)
                           │                  │       │          └ {'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlo...
                           │                  │       └ 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5'
                           │                  └ DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='App001', isP...
                           └ <function debug_walk_section at 0x0000028CBEB49EE0>

  File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\debug_engine.py", line 591, in debug_walk_section
    value=(dict(ctx.source_node) if ctx.source_node else None)
                │   │               │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                │   │               └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...
                │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...

ValueError: dictionary update sequence element #0 has length 1; 2 is required
2025-06-22 02:22:29.027 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-22 02:22:29.031 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-22 02:22:29.076 | ERROR    | app:debug_execute_flow:324 - Debug execution failed
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 120, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 352
               │     └ 3
               └ <function _main at 0x0000028C947D6480>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 133, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 352
           │    └ <function BaseProcess._bootstrap at 0x0000028C9427FC40>
           └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000028C9427F1A0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x0000028C9477EC10>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    └ <function subprocess_started at 0x0000028C96917BA0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x0000028C96A7C810>>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
           │       │   │    └ <function Server.serve at 0x0000028C96916C00>
           │       │   └ <uvicorn.server.Server object at 0x0000028C96A7C810>
           │       └ <function run at 0x0000028C9614A340>
           └ <module 'asyncio' from 'C:\\Program Files\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x0000028C96A808C0>
           │      └ <function Runner.run at 0x0000028C96264680>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\serve...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000028C96262200>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 637, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x0000028C96262160>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 604, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x0000028C96263F60>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000028C9612E520>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Program Files\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
          └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBEE340>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
          │    │               │      │        │                     └ Headers({'host': 'localhost:8005', 'connection': 'keep-alive', 'content-length': '170', 'sec-ch-ua-platform': '"Windows"', 'u...
          │    │               │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBEE340>
          │    │               │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │               └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function CORSMiddleware.simple_response at 0x0000028C98B604A0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
          │    │   │      │        └ functools.partial(<bound method CORSMiddleware.send of <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050...
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEDD00>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEDD00>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x0000028C941943D0>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEDD00>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │     └ <function Route.handle at 0x0000028C98AC84A0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEDD00>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function request_response.<locals>.app at 0x0000028CBEB76FC0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x0000028CBF0F9610>
                     └ <function get_request_handler.<locals>.app at 0x0000028CBEB76F20>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x0000028C98A92520>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='...
                 │         └ <function debug_execute_flow at 0x0000028CBEB76160>
                 └ <fastapi.dependencies.models.Dependant object at 0x0000028CBEBA9D50>

> File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\app.py", line 279, in debug_execute_flow
    response, debug_info = debug_walk_section(request.sectionId, ctx_dict)
                           │                  │       │          └ {'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlo...
                           │                  │       └ 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5'
                           │                  └ DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='App001', isP...
                           └ <function debug_walk_section at 0x0000028CBEB49EE0>

  File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\debug_engine.py", line 591, in debug_walk_section
    value=(dict(ctx.source_node) if ctx.source_node else None)
                │   │               │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                │   │               └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...
                │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...

ValueError: dictionary update sequence element #0 has length 1; 2 is required
2025-06-22 02:22:36.299 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-22 02:22:36.300 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-22 02:22:36.387 | ERROR    | app:debug_execute_flow:324 - Debug execution failed
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 120, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 352
               │     └ 3
               └ <function _main at 0x0000028C947D6480>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 133, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 352
           │    └ <function BaseProcess._bootstrap at 0x0000028C9427FC40>
           └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000028C9427F1A0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x0000028C9477EC10>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    └ <function subprocess_started at 0x0000028C96917BA0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x0000028C96A7C810>>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
           │       │   │    └ <function Server.serve at 0x0000028C96916C00>
           │       │   └ <uvicorn.server.Server object at 0x0000028C96A7C810>
           │       └ <function run at 0x0000028C9614A340>
           └ <module 'asyncio' from 'C:\\Program Files\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x0000028C96A808C0>
           │      └ <function Runner.run at 0x0000028C96264680>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\serve...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000028C96262200>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 637, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x0000028C96262160>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 604, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x0000028C96263F60>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000028C9612E520>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Program Files\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
          └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBEE0C0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
          │    │               │      │        │                     └ Headers({'host': 'localhost:8005', 'connection': 'keep-alive', 'content-length': '170', 'sec-ch-ua-platform': '"Windows"', 'u...
          │    │               │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBEE0C0>
          │    │               │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │               └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function CORSMiddleware.simple_response at 0x0000028C98B604A0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
          │    │   │      │        └ functools.partial(<bound method CORSMiddleware.send of <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050...
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBFDD00>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBFDD00>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x0000028C941943D0>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBFDD00>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │     └ <function Route.handle at 0x0000028C98AC84A0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBFDD00>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function request_response.<locals>.app at 0x0000028CBEB76FC0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x0000028CBF103ED0>
                     └ <function get_request_handler.<locals>.app at 0x0000028CBEB76F20>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x0000028C98A92520>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='...
                 │         └ <function debug_execute_flow at 0x0000028CBEB76160>
                 └ <fastapi.dependencies.models.Dependant object at 0x0000028CBEBA9D50>

> File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\app.py", line 279, in debug_execute_flow
    response, debug_info = debug_walk_section(request.sectionId, ctx_dict)
                           │                  │       │          └ {'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlo...
                           │                  │       └ 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5'
                           │                  └ DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='App001', isP...
                           └ <function debug_walk_section at 0x0000028CBEB49EE0>

  File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\debug_engine.py", line 591, in debug_walk_section
    value=(dict(ctx.source_node) if ctx.source_node else None)
                │   │               │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                │   │               └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...
                │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...

ValueError: dictionary update sequence element #0 has length 1; 2 is required
2025-06-22 02:22:42.775 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-22 02:22:42.944 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-22 02:22:43.011 | ERROR    | app:debug_execute_flow:324 - Debug execution failed
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 120, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 352
               │     └ 3
               └ <function _main at 0x0000028C947D6480>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 133, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 352
           │    └ <function BaseProcess._bootstrap at 0x0000028C9427FC40>
           └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000028C9427F1A0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x0000028C9477EC10>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    └ <function subprocess_started at 0x0000028C96917BA0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x0000028C96A7C810>>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
           │       │   │    └ <function Server.serve at 0x0000028C96916C00>
           │       │   └ <uvicorn.server.Server object at 0x0000028C96A7C810>
           │       └ <function run at 0x0000028C9614A340>
           └ <module 'asyncio' from 'C:\\Program Files\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x0000028C96A808C0>
           │      └ <function Runner.run at 0x0000028C96264680>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\serve...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000028C96262200>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 637, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x0000028C96262160>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 604, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x0000028C96263F60>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000028C9612E520>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Program Files\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
          └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBEDF80>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
          │    │               │      │        │                     └ Headers({'host': 'localhost:8005', 'connection': 'keep-alive', 'content-length': '170', 'sec-ch-ua-platform': '"Windows"', 'u...
          │    │               │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBEDF80>
          │    │               │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │               └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function CORSMiddleware.simple_response at 0x0000028C98B604A0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
          │    │   │      │        └ functools.partial(<bound method CORSMiddleware.send of <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050...
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEEFC0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEEFC0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x0000028C941943D0>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEEFC0>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │     └ <function Route.handle at 0x0000028C98AC84A0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEEFC0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function request_response.<locals>.app at 0x0000028CBEB76FC0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x0000028CBF0FBCD0>
                     └ <function get_request_handler.<locals>.app at 0x0000028CBEB76F20>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x0000028C98A92520>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='...
                 │         └ <function debug_execute_flow at 0x0000028CBEB76160>
                 └ <fastapi.dependencies.models.Dependant object at 0x0000028CBEBA9D50>

> File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\app.py", line 279, in debug_execute_flow
    response, debug_info = debug_walk_section(request.sectionId, ctx_dict)
                           │                  │       │          └ {'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlo...
                           │                  │       └ 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5'
                           │                  └ DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='App001', isP...
                           └ <function debug_walk_section at 0x0000028CBEB49EE0>

  File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\debug_engine.py", line 591, in debug_walk_section
    value=(dict(ctx.source_node) if ctx.source_node else None)
                │   │               │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                │   │               └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...
                │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...

ValueError: dictionary update sequence element #0 has length 1; 2 is required
2025-06-22 02:22:53.107 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-22 02:22:53.109 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-22 02:22:53.153 | ERROR    | app:debug_execute_flow:324 - Debug execution failed
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 120, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 352
               │     └ 3
               └ <function _main at 0x0000028C947D6480>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 133, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 352
           │    └ <function BaseProcess._bootstrap at 0x0000028C9427FC40>
           └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000028C9427F1A0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x0000028C9477EC10>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    └ <function subprocess_started at 0x0000028C96917BA0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x0000028C96A7C810>>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
           │       │   │    └ <function Server.serve at 0x0000028C96916C00>
           │       │   └ <uvicorn.server.Server object at 0x0000028C96A7C810>
           │       └ <function run at 0x0000028C9614A340>
           └ <module 'asyncio' from 'C:\\Program Files\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x0000028C96A808C0>
           │      └ <function Runner.run at 0x0000028C96264680>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\serve...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000028C96262200>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 637, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x0000028C96262160>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 604, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x0000028C96263F60>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000028C9612E520>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Program Files\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
          └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBEEE80>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
          │    │               │      │        │                     └ Headers({'host': 'localhost:8005', 'connection': 'keep-alive', 'content-length': '170', 'sec-ch-ua-platform': '"Windows"', 'u...
          │    │               │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBEEE80>
          │    │               │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │               └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function CORSMiddleware.simple_response at 0x0000028C98B604A0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
          │    │   │      │        └ functools.partial(<bound method CORSMiddleware.send of <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050...
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEE200>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEE200>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x0000028C941943D0>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEE200>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │     └ <function Route.handle at 0x0000028C98AC84A0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEE200>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function request_response.<locals>.app at 0x0000028CBEB76FC0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x0000028CBF0F2F90>
                     └ <function get_request_handler.<locals>.app at 0x0000028CBEB76F20>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x0000028C98A92520>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='...
                 │         └ <function debug_execute_flow at 0x0000028CBEB76160>
                 └ <fastapi.dependencies.models.Dependant object at 0x0000028CBEBA9D50>

> File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\app.py", line 279, in debug_execute_flow
    response, debug_info = debug_walk_section(request.sectionId, ctx_dict)
                           │                  │       │          └ {'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlo...
                           │                  │       └ 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5'
                           │                  └ DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='App001', isP...
                           └ <function debug_walk_section at 0x0000028CBEB49EE0>

  File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\debug_engine.py", line 591, in debug_walk_section
    value=(dict(ctx.source_node) if ctx.source_node else None)
                │   │               │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                │   │               └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...
                │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...

ValueError: dictionary update sequence element #0 has length 1; 2 is required
2025-06-23 02:52:36.765 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-23 02:52:36.782 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-23 02:52:36.891 | ERROR    | app:debug_execute_flow:324 - Debug execution failed
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 120, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 352
               │     └ 3
               └ <function _main at 0x0000028C947D6480>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 133, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 352
           │    └ <function BaseProcess._bootstrap at 0x0000028C9427FC40>
           └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000028C9427F1A0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x0000028C9477EC10>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    └ <function subprocess_started at 0x0000028C96917BA0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x0000028C96A7C810>>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
           │       │   │    └ <function Server.serve at 0x0000028C96916C00>
           │       │   └ <uvicorn.server.Server object at 0x0000028C96A7C810>
           │       └ <function run at 0x0000028C9614A340>
           └ <module 'asyncio' from 'C:\\Program Files\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x0000028C96A808C0>
           │      └ <function Runner.run at 0x0000028C96264680>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\serve...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000028C96262200>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 637, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x0000028C96262160>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 604, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x0000028C96263F60>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000028C9612E520>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Program Files\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBF...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
          └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBEFEC0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
          │    │               │      │        │                     └ Headers({'host': 'localhost:8005', 'connection': 'keep-alive', 'content-length': '170', 'sec-ch-ua-platform': '"Windows"', 'u...
          │    │               │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBEFEC0>
          │    │               │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │               └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function CORSMiddleware.simple_response at 0x0000028C98B604A0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
          │    │   │      │        └ functools.partial(<bound method CORSMiddleware.send of <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050...
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEE3E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEE3E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x0000028C941943D0>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEE3E0>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │     └ <function Route.handle at 0x0000028C98AC84A0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBEE3E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function request_response.<locals>.app at 0x0000028CBEB76FC0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x0000028CBF0FF210>
                     └ <function get_request_handler.<locals>.app at 0x0000028CBEB76F20>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x0000028C98A92520>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='...
                 │         └ <function debug_execute_flow at 0x0000028CBEB76160>
                 └ <fastapi.dependencies.models.Dependant object at 0x0000028CBEBA9D50>

> File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\app.py", line 279, in debug_execute_flow
    response, debug_info = debug_walk_section(request.sectionId, ctx_dict)
                           │                  │       │          └ {'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlo...
                           │                  │       └ 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5'
                           │                  └ DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='App001', isP...
                           └ <function debug_walk_section at 0x0000028CBEB49EE0>

  File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\debug_engine.py", line 591, in debug_walk_section
    value=(dict(ctx.source_node) if ctx.source_node else None)
                │   │               │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                │   │               └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...
                │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...

ValueError: dictionary update sequence element #0 has length 1; 2 is required
2025-06-23 02:52:43.458 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-23 02:52:43.462 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-23 02:52:43.480 | ERROR    | app:debug_execute_flow:324 - Debug execution failed
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 120, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 352
               │     └ 3
               └ <function _main at 0x0000028C947D6480>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 133, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 352
           │    └ <function BaseProcess._bootstrap at 0x0000028C9427FC40>
           └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000028C9427F1A0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x0000028C9477EC10>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    └ <function subprocess_started at 0x0000028C96917BA0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x0000028C96A7C810>>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
           │       │   │    └ <function Server.serve at 0x0000028C96916C00>
           │       │   └ <uvicorn.server.Server object at 0x0000028C96A7C810>
           │       └ <function run at 0x0000028C9614A340>
           └ <module 'asyncio' from 'C:\\Program Files\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x0000028C96A808C0>
           │      └ <function Runner.run at 0x0000028C96264680>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\serve...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000028C96262200>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 637, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x0000028C96262160>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 604, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x0000028C96263F60>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000028C9612E520>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Program Files\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBE...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBE...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBE...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
          └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEB75A80>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
          │    │               │      │        │                     └ Headers({'host': 'localhost:8005', 'connection': 'keep-alive', 'content-length': '170', 'sec-ch-ua-platform': '"Windows"', 'u...
          │    │               │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEB75A80>
          │    │               │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │               └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function CORSMiddleware.simple_response at 0x0000028C98B604A0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
          │    │   │      │        └ functools.partial(<bound method CORSMiddleware.send of <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050...
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEB75BC0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEB75BC0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x0000028C941943D0>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEB75BC0>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │     └ <function Route.handle at 0x0000028C98AC84A0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEB75BC0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function request_response.<locals>.app at 0x0000028CBEB76FC0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x0000028CBEC0DA10>
                     └ <function get_request_handler.<locals>.app at 0x0000028CBEB76F20>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x0000028C98A92520>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='...
                 │         └ <function debug_execute_flow at 0x0000028CBEB76160>
                 └ <fastapi.dependencies.models.Dependant object at 0x0000028CBEBA9D50>

> File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\app.py", line 279, in debug_execute_flow
    response, debug_info = debug_walk_section(request.sectionId, ctx_dict)
                           │                  │       │          └ {'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlo...
                           │                  │       └ 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5'
                           │                  └ DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='App001', isP...
                           └ <function debug_walk_section at 0x0000028CBEB49EE0>

  File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\debug_engine.py", line 591, in debug_walk_section
    value=(dict(ctx.source_node) if ctx.source_node else None)
                │   │               │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                │   │               └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...
                │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...

ValueError: dictionary update sequence element #0 has length 1; 2 is required
2025-06-23 03:16:09.128 | INFO     | app:debug_execute_flow:263 - Debug execution started
2025-06-23 03:16:09.135 | INFO     | debug_engine:debug_walk_section:558 - Debug engine invoked for section SEC_0f962e4d-a932-4958-9352-b54e0ef92be5 | params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlow': True}
2025-06-23 03:16:09.192 | ERROR    | app:debug_execute_flow:324 - Debug execution failed
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 120, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 352
               │     └ 3
               └ <function _main at 0x0000028C947D6480>
  File "C:\Program Files\Python311\Lib\multiprocessing\spawn.py", line 133, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 352
           │    └ <function BaseProcess._bootstrap at 0x0000028C9427FC40>
           └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000028C9427F1A0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Program Files\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x0000028C9477EC10>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
    │    └ <function subprocess_started at 0x0000028C96917BA0>
    └ <SpawnProcess name='SpawnProcess-1' parent=12748 started>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x0000028C96A7C810>>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=812, family=2, type=1, proto=6, laddr=('0.0.0.0', 8005)>]
           │       │   │    └ <function Server.serve at 0x0000028C96916C00>
           │       │   └ <uvicorn.server.Server object at 0x0000028C96A7C810>
           │       └ <function run at 0x0000028C9614A340>
           └ <module 'asyncio' from 'C:\\Program Files\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x0000028C96A808C0>
           │      └ <function Runner.run at 0x0000028C96264680>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\serve...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000028C96262200>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x0000028C968D44D0>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 637, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x0000028C96262160>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 604, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x0000028C96263F60>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Program Files\Python311\Lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000028C9612E520>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Program Files\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBE...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000028C96A7CF10>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBE...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028CBE...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
          └ <fastapi.applications.FastAPI object at 0x0000028CBEABE590>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBFEAC0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000028CBEA78890>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
          │    │               │      │        │                     └ Headers({'host': 'localhost:8005', 'connection': 'keep-alive', 'content-length': '170', 'sec-ch-ua-platform': '"Windows"', 'u...
          │    │               │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000028CBEBFEAC0>
          │    │               │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │               └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function CORSMiddleware.simple_response at 0x0000028C98B604A0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
          │    │   │      │        └ functools.partial(<bound method CORSMiddleware.send of <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050...
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000028CBEABE050>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBFE980>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000028CBEAF44D0>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBFE980>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x0000028C941943D0>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000028CBEAF6A90>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBFE980>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │     └ <function Route.handle at 0x0000028C98AC84A0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000028CBEBFE980>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000028...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8005), 'cl...
          │    └ <function request_response.<locals>.app at 0x0000028CBEB76FC0>
          └ APIRoute(path='/api/execute', name='debug_execute_flow', methods=['POST'])
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x0000028CBEC4AFD0>
                     └ <function get_request_handler.<locals>.app at 0x0000028CBEB76F20>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x0000028C98A92520>
  File "C:\Users\vinit\ModelTesting\.venv\Lib\site-packages\fastapi\routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
                 │         │      └ {'request': DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='...
                 │         └ <function debug_execute_flow at 0x0000028CBEB76160>
                 └ <fastapi.dependencies.models.Dependant object at 0x0000028CBEBA9D50>

> File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\app.py", line 279, in debug_execute_flow
    response, debug_info = debug_walk_section(request.sectionId, ctx_dict)
                           │                  │       │          └ {'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId': 'App001', 'isPrimaryFlo...
                           │                  │       └ 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5'
                           │                  └ DebugExecuteRequest(sectionId='SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', applicationId='Appl_123', applicantId='App001', isP...
                           └ <function debug_walk_section at 0x0000028CBEB49EE0>

  File "C:\Users\vinit\ModelTesting\flow_engine_project\debugUI\backend\debug_engine.py", line 591, in debug_walk_section
    value=(dict(ctx.source_node) if ctx.source_node else None)
                │   │               │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                │   │               └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...
                │   └ '4:163667f3-0018-494d-a3f3-415ecb90e442:116'
                └ DebugContext(input_params={'sectionId': 'SEC_0f962e4d-a932-4958-9352-b54e0ef92be5', 'applicationId': 'Appl_123', 'applicantId...

ValueError: dictionary update sequence element #0 has length 1; 2 is required
2025-06-23 11:55:28.954 | INFO     | app:lifespan:91 - Debug interface shutting down...
